### DISCUSSION TITLE
Problem while creating user group in portal. Upgraded portal from 2.6.0 to 3.1.0

### DISCUSSION THREAD

**Discussion by shakya-nischal** [pimcore/pimcore#16309] [Q&A / Support]
Compile Error: Declaration of Pimcore\Model\DataObject\PortalUserGroup::getAssetWorkspaceDefinition() must be compatible with Pimcore\Bundle\PortalEngineBundle\Model\DataObject\PortalUserGroupInterface::getAssetWorkspaceDef  
  inition(): ?array
  
  I am getting this error while creating portal user group in Pimcore 10.6

**Comment by kingjia90** [comment]
Thank you for reporting, opened a PR https://github.com/pimcore/portal-engine/pull/466 that may fix it.

However, just wanted to say that for enterprise bundles, it's not recommended to post here, please contact the enterprise support instead and also ask to get access to the private repository to check the patch.

**Reply by shakya-nischal** [reply]
Okay thankyou @kingjia90

**Comment by fashxp** [comment]
did you do following step before update: 
```
BEFORE UPGRADE: Activate Generate Type Declarations for PortalUser and PortalUserGroup classes and save them to regenerate the php class files. This will break portals until portal engine is upgraded to v3.0.0 via composer.
```
see also https://pimcore.com/docs/platform/Portal_Engine/Installation/Upgrade

**Reply by shakya-nischal** [reply]
Yes, I have activate generate type declaration for both potal user and group.

**Reply by fashxp** [reply]
and still the problem from above? did the regeneration of the object classes work?
